### PR TITLE
Chore: all jest to monorepo root

### PR DESCRIPTION
**Pull request by IanLondon**
Jest settings in project `package.json`s, and `make test` stuff in project `Makefile`s, is duplicated in every project.

## changelog
* all `"jest": {...}` settings and the `"jest"` + `"babel-jest"` dependencies moved up from projects to the monorepo's `package.json`
* `"scripts"` removed from all project `package.json`s (they're just aliases for calling `make`), except for `app-shell` which uses the `postinstall` hook and is unchanged
* `make lint-js` runs `flow check` not `flow status`, to eliminate the `Started a new flow server: -Please wait. Server is starting up: \Please wait. Server is initializing (starting up): |Please wait.` text in CI
* `make install-types` installs jest type defs
* `labware-definitions/` project also gets a `make install-types` command of its own (I didn't put one in yet)

## review requests

* (check CI) should not change code coverage
* Should not change *flow* coverage (eg missing flow-typed defs)
* Make sure it works on your computer

Also: let's talk DX. Do you all a convenient way to run flow server without `make lint-js`? Because if `make lint-js` is now using `flow check` instead of `flow` (aka `flow status`), it's gonna take a few minutes each run, which is not nice. But I think `flow check` is good for CI and `flow status` is good for dev. I usually use `npx flow` or `./node_modules/.bin/flow` @mcous @Kadee80

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -38,8 +38,11 @@ install:
 
 .PHONY: install-types
 install-types:
-	$(MAKE) -C $(COMPONENTS_DIR) install-types
+	flow-typed install jest@^20.0.4 --flowVersion=0.61.0 # TODO Ian 2018-03-15: use jest version in package.json
```
Comment (mcous) [inline Makefile]: Couldn't we just do `flow-typed install --flowVersion=0.61.0`?

File: package.json [hunk 2]
```diff
@@ -49,7 +48,6 @@
     "eslint-plugin-react": "^7.6.1",
     "eslint-plugin-standard": "^3.0.1",
     "flow-bin": "^0.66.0",
-    "flow-typed": "^2.2.3",
```
Comment (mcous) [inline package.json]: If we're going to call `flow-typed` from repo root this should not be removed

File: Makefile (hunk 1 above)
Comment (IanLondon) [reply Makefile]: we could, that would grab the other devDependencies in project root too, where elsewhere we're doing no devDependencies. I'm not sure why we were doing `--ignoreDeps=dev` before?

**Comment by IanLondon** [comment]
Also :-1: this is decreasing coverage right now

**Pull request merged**